### ISSUE TITLE
Adding nullptr to std::string

### ISSUE THREAD

**Issue by psorokovnin-at-aligntech** [boostorg/compute#861]
In https://github.com/boostorg/compute/blob/develop/include/boost/compute/detail/path.hpp#L31 std::getenv can return nullptr. It happens for $HOME in my case and produces a segmentation fault. I don't know the specification for operator+(const char*, const std::string&), but I know that construction of std::string from nullptr is undefined behaviour.

**Comment by chetanpandey1266** [comment]
Can you please assign this issue with me?

**Comment by musa66666** [comment]
No, it is not possible to directly add nullptr to an std::string in the Boost.Compute library or in standard C++.

The std::string class in C++ does not have a constructor or assignment operator that takes a nullptr parameter. Assigning a nullptr directly to an std::string would result in a compilation error.

Here's an example of code that would produce a compilation error:
std::string str = nullptr;  // Error: Cannot assign nullptr to std::string
If you have a specific use case or requirement that involves adding nullptr to an std::string, you would need to provide more details about the context and the desired behavior, as well as any relevant code examples or specific library functions you are working with.
